File: scripts/update_company_domains.py
#!/usr/bin/env python3
"""
Update existing companies in database with domain information from companies.json

This script updates the metadata field of existing companies to include the domain.
"""
import os
import json
from pathlib import Path
import psycopg2

# Load DATABASE_URL from environment
DATABASE_URL = os.getenv('DATABASE_URL')

# If not in environment, try to load from .env.local
if not DATABASE_URL:
    env_file = Path(__file__).parent.parent / '.env.local'
    if env_file.exists():
        with open(env_file, 'r') as f:
            for line in f:
                line = line.strip()
                if line and not line.startswith('#') and '=' in line:
                    key, value = line.split('=', 1)
                    if key.strip() == 'DATABASE_URL':
                        DATABASE_URL = value.strip().strip('"').strip("'")
                        break

COMPANIES_FILE = Path(__file__).parent.parent / 'companies.json'

def load_companies():
    """Load companies from companies.json"""
    with open(COMPANIES_FILE, 'r') as f:
        data = json.load(f)
    return data['companies']

def update_company_domains(conn):
    """Update company metadata with domain information"""
    companies = load_companies()

    print(f"📋 Updating {len(companies)} companies with domain information")
    print()

    cursor = conn.cursor()

    # Update each company's metadata to include the domain
    update_count = 0
    for company in companies:
        ticker = company['ticker']
        domain = company.get('domain')

        if not domain:
            print(f"   ⚠️  No domain for {ticker}, skipping")
            continue

        # Update the metadata field with the complete company object
        cursor.execute(
            """
            UPDATE companies
            SET metadata = %s
            WHERE ticker = %s
            """,
            (json.dumps(company), ticker)
        )

        if cursor.rowcount > 0:
            update_count += 1
            print(f"   ✓ Updated {ticker} with domain {domain}")

    conn.commit()

    print()
    print(f"✅ Successfully updated {update_count} companies!")

    cursor.close()
    return update_count

def main():
    """Main update function"""
    print("="* 60)
    print("10KAY Company Domain Updater")
    print("="* 60)
    print()

    if not DATABASE_URL:
        print("✗ DATABASE_URL not found in environment variables")
        import sys
        sys.exit(1)

    if not COMPANIES_FILE.exists():
        print(f"✗ companies.json not found at {COMPANIES_FILE}")
        import sys
        sys.exit(1)

    try:
        # Connect to database
        print("🔌 Connecting to database...")
        conn = psycopg2.connect(DATABASE_URL)
        print("✓ Connected")
        print()

        # Update companies
        count = update_company_domains(conn)

        print()
        print("="* 60)
        print(f"✅ Update complete! Updated {count} companies.")
        print("="* 60)

    except psycopg2.Error as e:
        print(f"✗ Database error: {e}")
        import sys
        sys.exit(1)
    except Exception as e:
        print(f"✗ Unexpected error: {e}")
        import traceback
        traceback.print_exc()
        import sys
        sys.exit(1)
    finally:
        if 'conn' in locals():
            conn.close()

if __name__ == '__main__':
    main()

File: scripts/update-company-domains.ts
/**
 * Update existing companies in database with domain information from companies.json
 *
 * This script updates the metadata field of existing companies to include the domain.
 * Run with: npx tsx scripts/update-company-domains.ts
 */

import { readFileSync } from 'fs';
import { join } from 'path';
import pg from 'pg';

const { Pool } = pg;

interface Company {
  ticker: string;
  name: string;
  domain?: string;
  enabled: boolean;
}

interface CompaniesData {
  companies: Company[];
}

async function main() {
  console.log('='.repeat(60));
  console.log('10KAY Company Domain Updater');
  console.log('='.repeat(60));
  console.log();

  // Load companies.json
  const companiesPath = join(process.cwd(), 'companies.json');
  const companiesData: CompaniesData = JSON.parse(readFileSync(companiesPath, 'utf-8'));
  const companies = companiesData.companies;

  console.log(`📋 Updating ${companies.length} companies with domain information`);
  console.log();

  // Connect to database
  const pool = new Pool({
    connectionString: process.env.DATABASE_URL,
  });

  try {
    console.log('🔌 Connecting to database...');
    const client = await pool.connect();
    console.log('✓ Connected');
    console.log();

    let updateCount = 0;

    for (const company of companies) {
      const { ticker, domain } = company;

      if (!domain) {
        console.log(`   ⚠️  No domain for ${ticker}, skipping`);
        continue;
      }

      // Update the metadata field with the complete company object
      const result = await client.query(
        `UPDATE companies
         SET metadata = $1
         WHERE ticker = $2`,
        [JSON.stringify(company), ticker]
      );

      if (result.rowCount && result.rowCount > 0) {
        updateCount++;
        console.log(`   ✓ Updated ${ticker} with domain ${domain}`);
      }
    }

    client.release();

    console.log();
    console.log(`✅ Successfully updated ${updateCount} companies!`);
    console.log();
    console.log('='.repeat(60));
    console.log(`✅ Update complete! Updated ${updateCount} companies.`);
    console.log('='.repeat(60));

  } catch (error) {
    console.error('✗ Error:', error);
    process.exit(1);
  } finally {
    await pool.end();
  }
}

main();
